src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentsService.java

```diff
@@ -129,4 +153,38 @@ private Map<String, String> makeLocaleMap(final SortedSet<String> localesToQuery
 
         return map;
     }
+
+    private void getFallbackLocaleMessages(Iterator<Locale> fallbackLocalesIter, final Set<String> componentsToFallback,
+                                                                       final Set<String> localesToFallback, final Map<String, Map<String, Map<String, String>>> dataMap) {
+        if (fallbackLocalesIter != null && fallbackLocalesIter.hasNext()) {
+            Locale fallbackLocale = fallbackLocalesIter.next();
+            if (fallbackLocale.toLanguageTag().equals(ConstantsKeys.SOURCE)) {
+                fallbackLocale = Locale.forLanguageTag(ConstantsKeys.LATEST);
+            }
+            Set<Locale> localeSet = new HashSet<Locale>();
+            localeSet.add(fallbackLocale);
+            Map<Locale, Map<String, Map<String, String>>> localeDataMap = getTranslation(componentsToFallback, localeSet, fallbackLocalesIter);
+            Map<String, Map<String, String>> componentsMap = localeDataMap.get(fallbackLocale);
+            if (componentsMap != null && !componentsMap.isEmpty()) {
+                final SortedSet<String> localesToFallbackAgain = new TreeSet<>();
+                final SortedSet<String> componentsToFallbackAgain = new TreeSet<>();
```

Comment:
This is unnecessary because it has been handled in 119-120.

Reply:
Here doesn't do fallback again and the variables are just for logging the fallback failure message. Rename the variables' name to make code more readable.